src/box/iproto.cc

```diff
@@ -258,6 +266,23 @@ unsigned iproto_readahead = 16320;
 /* The maximal number of iproto messages in fly. */
 static int iproto_msg_max = IPROTO_MSG_MAX_MIN;
 
+#define mh_name _rhvt
+#define mh_key_t uint32_t
+#define mh_node_t struct iproto_handler
```

Comment:
Because using the pointer requires dynamic allocation of the _value_ structure.

Reply:
Why do you think dynamic allocations are unacceptable in this case?

Reply:
This not a problem of dynamic allocation, but rather a problem of encapsulation. With my approach, I delegate memory management of hash table nodes to the hash table — AFAIC, what you suggest means we have to manage the memory of hash table nodes ourselves.

Reply:
Discussed with @locker and @alyapunov: decided to use `mh_i32ptr_t`, since we have to manage the request handler lifetime anyways (i.e., call its destructor).